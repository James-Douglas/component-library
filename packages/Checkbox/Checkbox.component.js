import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons/faCheck';
import { usePrefill, useId } from '@comparethemarketau/manor-hooks';
import { FieldValidation } from '@comparethemarketau/manor-field-validation';

import {
  StyledCheckbox,
  StyledContent,
  StyledHiddenInput,
  StyledLabel,
  StyledWrap,
} from './Checkbox.styles';

export const renderIcon = (icon, value) => {
  if (value) {
    return (
      <FontAwesomeIcon icon={icon} size="1x" />
    );
  }
  return null;
};

export const getInitialValue = (isSelected, prefillValue) => isSelected || prefillValue || false;

const Checkbox = ({
  id: propsId,
  icon,
  label,
  disabled,
  validationMessage,
  invertColour,
  handleClick,
  handleChange,
  handleFocus,
  handleBlur,
  isSelected,
  prefillValue,
  className,
}) => {
  const [internalValue, setInternalValue] = useState(getInitialValue(isSelected, prefillValue));
  const [isDirty, setIsDirty] = useState(false);
  const isAutofill = usePrefill(prefillValue, isSelected, isDirty);
  const id = useId(propsId);

  useEffect(() => {
    if (!prefillValue) {
      setInternalValue(isSelected);
    }
  }, [isSelected, prefillValue]);

  const toggleEventHandler = (e) => {
    const targValue = e.target.checked;

    setIsDirty(true);
    setInternalValue(targValue);

    if (handleChange) {
      handleChange(id, targValue);
    }
  };

  return (
    <>
      <StyledWrap className={className}>
        <StyledHiddenInput
          id={id}
          name={id}
          type="checkbox"
          role="checkbox"
          disabled={disabled}
          onClick={handleClick}
          onChange={toggleEventHandler}
          checked={internalValue}
          aria-checked={internalValue}
          onFocus={handleFocus}
          onBlur={handleBlur}
        />
        <StyledLabel
          disabled={disabled}
          htmlFor={id}
        >
          <StyledCheckbox
            invertColour={invertColour}
            checked={internalValue}
            isAutofill={isAutofill}
            disabled={disabled}
            invalid={validationMessage && validationMessage.length > 0}
          >
            {renderIcon(icon, internalValue)}
          </StyledCheckbox>
          {label
            && (
              <StyledContent>
                {label}
              </StyledContent>
            )}
        </StyledLabel>
      </StyledWrap>
      <FieldValidation message={validationMessage} />
    </>
  );
};

Checkbox.propTypes = {
  /**
   * Unique identifier for the checkbox.
   */
  id: PropTypes.string,
  /**
   * The label for the Checkbox
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /**
   * Defines the icon needed for the checkbox.
   */
  icon: PropTypes.oneOfType([
    PropTypes.shape({
      prefix: PropTypes.string,
      iconName: PropTypes.string,
      icon: PropTypes.array, // eslint-disable-line
    }),
    PropTypes.string,
  ]),
  /**
   * Defines if the checkbox is disabled or not.
   */
  disabled: PropTypes.bool,
  /**
   * Defines the checkbox checked state.
   */
  isSelected: PropTypes.bool,
  /**
   * Prefills the checkbox and applies browser autocomplete styles
   */
  prefillValue: PropTypes.bool,
  /**
   * Displays given validation message and invalid styles on the component when provided.
   */
  validationMessage: PropTypes.string,
  /**
   * Changes the color of the checked state - blue bg, white tick, or white bg, blue tick.
   */
  invertColour: PropTypes.bool,
  /**
   * Called on click
   */
  handleClick: PropTypes.func,
  /**
   * Called on change with { id, value }.
   */
  handleChange: PropTypes.func,
  /**
   * Called on focus of the checkbox
   */
  handleFocus: PropTypes.func,
  /**
   * Called on blur of the checkbox
   */
  handleBlur: PropTypes.func,
  /**
   * Classes to be applied to the Checkbox component
   */
  className: PropTypes.string,
};

Checkbox.defaultProps = {
  id: null,
  icon: faCheck,
  label: null,
  disabled: false,
  isSelected: false,
  prefillValue: false,
  validationMessage: '',
  invertColour: false,
  handleClick: null,
  handleChange: null,
  handleFocus: null,
  handleBlur: null,
  className: '',
};

export default Checkbox;
