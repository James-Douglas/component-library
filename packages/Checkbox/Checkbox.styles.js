import styled, { css } from 'styled-components';

export const StyledHiddenInput = styled.input`
  border: 0;
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: ${({ theme }) => theme.spacing.px};
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: ${({ theme }) => theme.spacing.px};
`;

export const StyledWrap = styled.div`
  ${({ disabled, theme, variant }) => {
    if (variant === 'boxed') {
      if (!disabled) {
        return css`
        border: 1px solid ${theme.colors.grey100};
        padding: ${theme.spacing[8]} ${theme.spacing[0]} ${theme.spacing[12]} ${theme.spacing[16]};
        box-shadow: ${theme.checkbox.boxedShadow};
        :hover {
          background: ${theme.colors.primary50};
          border: ${theme.checkbox.borderHover};
          cursor: pointer;
        }
        :active, :focus {
          border: ${theme.checkbox.borderHover};
        }
        :hover ${StyledCheckbox}, :active ${StyledCheckbox}, :focus ${StyledCheckbox} {
          border: ${theme.checkbox.borderHover};
        }
        margin-bottom: ${theme.spacing[12]};
      `;
      }
      return css`
        cursor: not-allowed;
        padding: ${theme.spacing[8]} ${theme.spacing[0]} ${theme.spacing[12]} ${theme.spacing[16]};
        background: ${theme.checkbox.backgroundDisabled};
        border: 1px solid ${theme.colors.grey100};
        color: ${theme.checkbox.colorDisabled};
        opacity:${theme.checkbox.disabledOpacity};
        margin-bottom: ${theme.spacing[12]};
      `;
    }
    return css`  
    display: flex;
    margin-bottom: ${theme.spacing[12]};
    min-width: 3rem;
    `;
  }}
  ${({ isFocused, theme, variant }) => (isFocused && variant === 'boxed') && css`
  border: ${theme.checkbox.borderHover};
  `}
`;

export const StyledCheckbox = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border: ${({ theme }) => theme.checkbox.border};
  ${({ theme, checked, isAutofill }) => {
    if (isAutofill) {
      return css`
        background: ${theme.checkbox.prefilled};
        border: ${theme.checkbox.prefilledBorder};
      `;
    } if (checked) {
      return css`
        background: ${theme.checkbox.backgroundChecked};
      `;
    }
    return css`
      background: ${theme.checkbox.background};
    `;
  }}

  color: ${({ theme, isAutofill }) => (isAutofill ? theme.checkbox.iconPrefilled : theme.checkbox.color)};
  min-width: ${({ theme }) => theme.checkbox.size};
  height: ${({ theme }) => theme.checkbox.size};
  pointer-events: none;
  border-radius: ${({ theme }) => theme.checkbox.borderRadius};
  padding: ${({ theme }) => theme.spacing.px};

  ${StyledHiddenInput}:focus + label & {
    box-shadow: ${({ theme }) => theme.checkbox.shadow};
  }

  ${({ invertColour, theme }) => invertColour && css`
    background: ${theme.checkbox.background};
    color: ${theme.checkbox.backgroundChecked};
  `}

  ${({ disabled, theme }) => disabled && css`
    pointer-events: none;
    background: ${theme.checkbox.backgroundDisabled};
    border: ${theme.borders.disabled};
    color: ${theme.checkbox.colorDisabled};
    opacity:${theme.checkbox.disabledOpacity};
  `}
  ${({ disabled, theme, variant }) => disabled && variant === 'boxed' && css`
    border: 1px solid ${theme.colors.grey500};
  `}
  ${({ theme, invalid }) => invalid && css`
    border: ${theme.checkbox.invalid};
    :hover {
      border: ${theme.borders.invalid};
    }
  `}
`;

export const StyledLabel = styled.label`
  font-family: ${({ theme }) => theme.fontFamily};
  font-weight: ${({ theme }) => theme.fontWeight.normal};
  color: ${({ theme }) => theme.colors.grey900};
  display: flex;
  align-items: flex-start;
  cursor: pointer;
  margin-top: ${({ theme }) => theme.spacing[4]};

  ${({ disabled }) => disabled && css`
    cursor: not-allowed;
  `};
  :hover ${StyledCheckbox} {
    border: ${({ theme, disabled }) => (disabled ? null : theme.checkbox.borderHover)};
  }
  :active ${StyledCheckbox} {
   border: ${({ theme, disabled }) => (disabled ? null : theme.checkbox.borderHover)};
  }
`;

export const StyledContent = styled.div`
  margin: ${({ theme }) => theme.checkbox.contentMargin};
  pointer-events: none;
  font-size: ${({ theme }) => theme.fontSize.base};
  line-height: ${({ theme }) => theme.spacing[24]};
`;

export const StyledImg = styled.img`
  min-width: ${({ theme }) => theme.spacing[24]};
  height: ${({ theme }) => theme.spacing[24]};
  margin-left: ${({ theme }) => theme.spacing[8]};
`;
