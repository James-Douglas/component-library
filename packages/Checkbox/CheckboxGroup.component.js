import React, {
  useState, useEffect, useRef, useCallback, useContext,
} from 'react';
import PropTypes from 'prop-types';
import { ManorContext } from '@comparethemarketau/manor-provider';
import { useId } from '@comparethemarketau/manor-hooks';
import { FieldValidation } from '@comparethemarketau/manor-field-validation';
import { Row, Column } from '@comparethemarketau/manor-grid';
import { Label } from '@comparethemarketau/manor-label';
import { tooltipPropTypes } from '@comparethemarketau/manor-tooltip';
import { StyledContainer, StyledInnerRow } from './CheckboxGroup.styles';

export const generateGroup = (colSize, children, callback, selectedCheckboxes) => {
  if (children) {
    return children.map((child) => {
      const component = React.cloneElement(child, { handleChange: callback, isSelected: selectedCheckboxes.includes(child.props.id) });
      return (
        <Column cols={colSize} key={`key-${child.props.id}`}>
          {component}
        </Column>
      );
    });
  }
  return null;
};

const CheckboxGroup = ({
  label,
  tooltip,
  validationMessage,
  groupId: propsGroupId,
  colSize,
  handleChange,
  selected,
  children,
}) => {
  const groupId = useId(propsGroupId);
  const [selectedCheckboxes, setSelectedCheckboxes] = useState(selected);
  const { isDesktop } = useContext(ManorContext);
  const didMount = useRef(false);

  const handleCheckboxClick = useCallback((id) => {
    const exists = selectedCheckboxes.includes(id);
    if (!exists) {
      const newSelectedCheckboxes = [...selectedCheckboxes, id];
      setSelectedCheckboxes(newSelectedCheckboxes);
    } else {
      const newSelectedCheckboxes = selectedCheckboxes.filter((selectedCheckbox) => selectedCheckbox !== id);
      setSelectedCheckboxes(newSelectedCheckboxes);
    }
  }, [selectedCheckboxes]);

  useEffect(() => {
    if (didMount.current) {
      handleChange(selectedCheckboxes);
    } else {
      didMount.current = true;
    }
  }, [handleChange, selectedCheckboxes]);

  return (
    <>
      <Label htmlFor={groupId} text={label} tooltip={tooltip} />
      <Row removeMarginBottom>
        <Column cols={isDesktop ? 10 : 12}>
          <StyledContainer id={groupId}>
            <StyledInnerRow>
              {generateGroup(colSize, children, handleCheckboxClick, selectedCheckboxes)}
            </StyledInnerRow>
          </StyledContainer>
          <FieldValidation message={validationMessage} />
        </Column>
      </Row>
    </>
  );
};

CheckboxGroup.propTypes = {
  /**
   * Unique identifier for the CheckboxGroup.
   */
  groupId: PropTypes.string,
  /**
   * Label for the CheckboxGroup.
   */
  label: PropTypes.oneOf([PropTypes.string, PropTypes.node]),
  /**
   * Tooltip object (see Tooltip documentation)
   */
  tooltip: PropTypes.shape(tooltipPropTypes),
  /**
   * Displays given validation message and invalid styles on the component when provided.
   */
  validationMessage: PropTypes.string,
  /**
   * Defines the sizing of the columns to wrap the checkboxes in.
   */
  colSize: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
  /**
   * Called whenever a checkbox is selected or deselected. It is called with an array of ids representing the currently
   selected checkboxes)
   */
  handleChange: PropTypes.func,
  /**
   * Sets the selected checkboxes
   */
  selected: PropTypes.arrayOf(PropTypes.string),
  /**
   * The child Checkbox components to render in the CheckboxGroup
   */
  children: (props, propname, componentName) => componentName === 'Checkbox',
};

CheckboxGroup.defaultProps = {
  groupId: null,
  label: '',
  tooltip: {},
  validationMessage: null,
  colSize: 6,
  handleChange: () => { },
  selected: [],
  children: [],
};

export default CheckboxGroup;
