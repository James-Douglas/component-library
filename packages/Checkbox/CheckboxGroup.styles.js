import styled from 'styled-components';
import { Row } from '@comparethemarketau/manor-grid';

export const StyledContainer = styled.div`
  width: 100%;
`;

export const StyledInnerRow = styled(Row)`
  margin: 0;
`;
