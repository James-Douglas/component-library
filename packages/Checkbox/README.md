# Checkbox

## Installation

`yarn add @comparethemarketau/manor-checkbox`


## Documentation

For technical documentation, see [Storybook](https://services.dev.comparethemarket.cloud/manor/?path=/docs/components-checkbox--checkbox).
Design system documentation can be found here [Bento](https://zeroheight.com/9942937b5/p/212a26-checkboxes/b/97d6bd)
