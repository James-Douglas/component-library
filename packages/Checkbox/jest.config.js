const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Checkbox',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Checkbox',
  displayName: 'Checkbox',
  rootDir: '../..',
};
