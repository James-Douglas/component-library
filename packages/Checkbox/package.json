{
  "name": "@comparethemarketau/manor-checkbox",
  "version": "0.5.5",
  "description": "Manor Checkbox component",
  "main": "lib/index.cjs.js",
  "author": "Compare The Market",
  "license": "Proprietary",
  "scripts": {
    "build": "npx rollup -c '../../rollup.config.js'",
    "tdd": "jest --coverage --watchAll",
    "test": "cross-env JEST_SILENT_REPORTER_DOTS=true jest --reporters=jest-silent-reporter --coverage --watchAll",
    "test:ci": "cross-env CI=true jest",
    "prepare": "cp package.json lib/ && cp README.md lib/ && cp CHANGELOG.md lib/"
  },
  "publishConfig": {
    "registry": "https://nexus.comparethemarket.com.au/repository/platform-packages/",
    "access": "public",
    "tag": "latest"
  },
  "dependencies": {
    "@babel/runtime": "^7.0.0",
    "@comparethemarketau/manor-field-validation": "^0.4.4",
    "@comparethemarketau/manor-grid": "^0.6.5",
    "@comparethemarketau/manor-hooks": "^0.2.1",
    "@comparethemarketau/manor-label": "^0.3.4",
    "@comparethemarketau/manor-tooltip": "^0.5.4",
    "@fortawesome/fontawesome-svg-core": "^1.2.26",
    "@fortawesome/free-solid-svg-icons": "^5.12.0",
    "@fortawesome/pro-regular-svg-icons": "^5.12.0",
    "@fortawesome/react-fontawesome": "^0.1.8",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@comparethemarketau/manor-themes": "^0.5.0"
  },
  "peerDependencies": {
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "styled-components": "^5.0.0"
  }
}
