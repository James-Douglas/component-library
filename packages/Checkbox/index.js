import Checkbox from './Checkbox.component';
import CheckboxGroup from './CheckboxGroup.component';

export { Checkbox, CheckboxGroup };
