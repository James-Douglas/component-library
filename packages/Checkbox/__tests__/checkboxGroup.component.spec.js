import React from 'react';
import { ctmTheme } from '@comparethemarketau/manor-themes';
import { render, fireEvent } from '../../../testUtils';
import CheckboxGroup, { generateGroup } from '../CheckboxGroup.component';
import Checkbox from '../Checkbox.component';
import sergei from '../../../images/sergei.png';

describe('generateGroup', () => {
  // eslint-disable-next-line react/prop-types
  const GroupContainer = ({ colSize, children }) => (
    <>
      {generateGroup(colSize, children, null, [])}
    </>
  );

  it('does not render children if there are none', () => {
    const { container } = render(<GroupContainer />);
    expect(container).toMatchSnapshot();
  });

  it('renders if there are children', () => {
    const { container } = render(
      <GroupContainer colSize={6}>
        <Checkbox id="A-1" label="A-1 check" />
        <Checkbox id="A-2" label="A-2 check" />
      </GroupContainer>,
    );
    const allLabel = container.querySelectorAll('label');
    expect(allLabel.length).toBe(2);
    expect(container.innerHTML).toMatchSnapshot();
  });
});

describe('CheckboxGroup.component', () => {
  it('renders with minimal props', () => {
    const { container } = render(
      <CheckboxGroup label="test" groupId="test-group-id">
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3"><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders with props', () => {
    const mockTestClick = jest.fn();
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5} handleChange={mockTestClick}>
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3"><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );

    const checkboxGroup = container.querySelector('#test-group-id');
    const chkA1 = container.querySelector('label[for="A-1"]');
    const chkA1Style = chkA1.firstChild;
    const inputId = container.querySelector('#A-1');

    fireEvent.click(chkA1, { button: 0 });
    expect(checkboxGroup.id).toBe('test-group-id');
    expect(chkA1Style).toHaveStyle(`background: ${ctmTheme.checkbox.backgroundChecked}`);
    expect(chkA1.getAttribute('disabled')).toBe(null);
    expect(chkA1.getElementsByTagName('svg')).toBeDefined();
    expect(mockTestClick).toHaveBeenCalled();
    expect(inputId.checked).toBe(true);
    fireEvent.click(chkA1, { button: 0 });
    expect(inputId.checked).toBe(false);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('checks on click', () => {
    const mockTestClick = jest.fn();
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5} handleClick={mockTestClick}>
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3"><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );

    const chkA1Input = container.querySelector('#A-1');

    const chkA1Label = container.querySelector('label[for="A-1"]');
    fireEvent.click(chkA1Label, { button: 0 });

    expect(chkA1Input.checked).toBe(true);
  });

  it('accepts a prefill value', () => {
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5} selected={['A-1', 'A-3']}>
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3"><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );

    const checkboxA1 = container.querySelector('#A-1');
    const checkboxA2 = container.querySelector('#A-2');
    const checkboxA3 = container.querySelector('#A-3');
    expect(checkboxA1.checked).toBe(true);
    expect(checkboxA2.checked).toBe(false);
    expect(checkboxA3.checked).toBe(true);
  });

  it('does not check when disabled', () => {
    const mockTestClick = jest.fn();
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5} handleClick={mockTestClick}>
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3" disabled><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );

    const chkA3Input = container.querySelector('#A-3');

    const chkA3Label = container.querySelector('label[for="A-3"]');
    fireEvent.click(chkA3Label, { button: 0 });

    expect(chkA3Input.checked).toBe(false);
  });

  it('render with prop tooltip', () => {
    const tooltip = {
      title: 'Tooltip heading',
      body: 'Prefix and suffix view',
    };
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5} tooltip={tooltip}>
        <Checkbox id="A-1"><p>A-1 check</p></Checkbox>
        <Checkbox id="A-2"><p>A-2 check</p></Checkbox>
        <Checkbox id="A-3"><p>A-3 check</p></Checkbox>
      </CheckboxGroup>,
    );
    const tooltipExist = container.querySelector('div[role="tooltip"]');
    expect(tooltipExist).toBeInTheDocument();
  });

  it('render with img', () => {
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5}>
        <Checkbox id="A-1" image={sergei}>A-1 check</Checkbox>
        <Checkbox id="A-2">A-2 check</Checkbox>
        <Checkbox id="A-3">A-3 check</Checkbox>
      </CheckboxGroup>,
    );
    const chk1Img = container.querySelector('label[for="A-1"] img');
    expect(chk1Img).toBeInTheDocument();
    expect(chk1Img).toHaveAttribute('src', 'sergei.png');
    const chk2Img = container.querySelector('label[for="A-2"] img');
    expect(chk2Img).not.toBeInTheDocument();
  });

  it('renders boxed checkboxes', () => {
    const { container } = render(
      <CheckboxGroup groupId="test-group-id" colSize={5}>
        <Checkbox id="A-1" variant="boxed">A-1 check</Checkbox>
        <Checkbox id="A-2">A-2 check</Checkbox>
        <Checkbox id="A-3">A-3 check</Checkbox>
      </CheckboxGroup>,
    );
    const chkA1Div = container.querySelector('#A-1').closest('div');
    expect(chkA1Div).toHaveStyle(`border: 1px solid ${ctmTheme.colors.grey100}`);
    const chkA2Div = container.querySelector('#A-2').closest('div');
    expect(chkA2Div).not.toHaveStyle(`border: 1px solid ${ctmTheme.colors.grey100}`);
  });
});
