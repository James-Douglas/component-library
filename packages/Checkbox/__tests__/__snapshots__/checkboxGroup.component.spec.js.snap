// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup.component renders with minimal props 1`] = `
"<div class=\\"sc-fzoXzr dgQelt\\"><label for=\\"test-group-id\\" class=\\"sc-fzpjYC gUAKL label\\">test</label></div><div class=\\"sc-AxhUy sc-fzozJi hRZogA row \\"><div class=\\"sc-AxgMl sc-AxheI gsRwqP 
        
      \\"><div id=\\"test-group-id\\" class=\\"sc-fznxsB gJmhvS\\"><div class=\\"sc-AxhUy sc-fzozJi fBSpRK row sc-fznJRM eExOsy\\"><div class=\\"sc-AxgMl sc-AxheI igotej 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-1\\" name=\\"A-1\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-1\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div><div class=\\"sc-AxgMl sc-AxheI igotej 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-2\\" name=\\"A-2\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-2\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div><div class=\\"sc-AxgMl sc-AxheI igotej 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-3\\" name=\\"A-3\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-3\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div></div></div></div></div>"
`;

exports[`CheckboxGroup.component renders with props 1`] = `
"<div class=\\"sc-fzoXzr dgQelt\\"><label for=\\"test-group-id\\" class=\\"sc-fzpjYC gUAKL label\\"></label></div><div class=\\"sc-AxhUy sc-fzozJi hRZogA row \\"><div class=\\"sc-AxgMl sc-AxheI gsRwqP 
        
      \\"><div id=\\"test-group-id\\" class=\\"sc-fznxsB gJmhvS\\"><div class=\\"sc-AxhUy sc-fzozJi fBSpRK row sc-fznJRM eExOsy\\"><div class=\\"sc-AxgMl sc-AxheI ifRcLL 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-1\\" name=\\"A-1\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-1\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div><div class=\\"sc-AxgMl sc-AxheI ifRcLL 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-2\\" name=\\"A-2\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-2\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div><div class=\\"sc-AxgMl sc-AxheI ifRcLL 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-3\\" name=\\"A-3\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-3\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div></label></div></div></div></div></div></div>"
`;

exports[`generateGroup does not render children if there are none 1`] = `<div />`;

exports[`generateGroup renders if there are children 1`] = `
"<div class=\\"sc-AxgMl sc-AxheI igotej 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-1\\" name=\\"A-1\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-1\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div><div class=\\"sc-fzoyTs fXOGpI\\">A-1 check</div></label></div></div><div class=\\"sc-AxgMl sc-AxheI igotej 
        
      \\"><div class=\\"sc-fzoiQi kojvwE\\"><input id=\\"A-2\\" name=\\"A-2\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-fznWqX kVHZaY\\"><label for=\\"A-2\\" class=\\"sc-fzqNqU cLRsou\\"><div class=\\"sc-fzqARJ fqUSrT\\"></div><div class=\\"sc-fzoyTs fXOGpI\\">A-2 check</div></label></div></div>"
`;
